source/fundamentals/crud/read-operations/project.txt

```diff
@@ -0,0 +1,153 @@
+==============================
+Specify Which Fields to Return
+==============================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+In this guide, you will learn how to control which fields appear in
+documents returned by read operations.
+
+Many read requests require only a subset of fields in a document.
+For example, when logging a user in you may only need their username, and
+not all of their profile information. By default, queries in MongoDB return
+all fields in matching documents. You can use a **projection** to return
+only the data you need.
+
+A projection is a document sent that instructs MongoDB which fields of a
+document to return. Use the :ref:`Projections <projections-builders>` class
+to easily construct a projection document.
+
+Behavior
+--------
+
+Projections work in two ways:
+
+- Explicitly including fields. This has the side-effect of implicitly
+  excluding all unspecified fields.
+
+- Implicitly excluding fields. This has the side-effect of implicitly
+  including all unspecified fields.
+
+These two methods of projection are mutually exclusive: if you
+explicitly include fields, you cannot explicitly exclude fields, and
+vice versa.
+
+.. important::
+
+   The ``_id`` field is **not** subject to these mechanics. You must
+   explicitly exlude the ``_id`` field if you do not want it returned.
+   You can exclude the ``_id`` field even if you have specified certain
+   fields to include.
+
+   .. code-block:: java
+
+      .projection(Projections.fields(Projections.include("username"), Projections.ecludeId()))
+
+Explanation
+-----------
+
+Consider the following collection containing documents that describe
+varieties of fruit:
+
+.. code-block:: json
+
+     { "_id": 1, "name": "apples", "qty": 5, "rating": 3 },
+     { "_id": 2, "name": "bananas", "qty": 7, "rating": 1 },
+     { "_id": 3, "name": "oranges", "qty": 6, "rating": 2 },
+     { "_id": 4, "name": "avocados", "qty": 3, "rating": 5 },
+
+In the following query, pass the projection to only return the ``name``
+field of each document:
+
+.. code-block:: java
+
+   // return all documents with only* the name field
+   collection.find(Filters.empty())
+       .projection(Projections.fields(Projections.include("name")))
+       .forEach(doc -> System.out.println(doc));
+
+
+The projection document specifies that the read operation result should
+*include* the ``name`` field of each returned document. As a result, this
+projection implicitly excludes the ``qty`` and ``rating`` fields. Chaining
+this projection to ``find()`` with an empty query document and no sort
+document yields the following results:
```

Comment:
Question: Why is "no sort document" being mentioned when this page isn't about sorts?

Reply:
Good question. I'll remove it!